kong/db/init.lua

```diff
@@ -65,8 +66,12 @@ function DB.new(kong_config, strategy)
         return nil, fmt("schema of entity '%s' is invalid: %s", entity_name,
                         tostring(errors:schema_violation(err_t)))
       end
-
-      schemas[entity_name] = Entity.new(entity_schema)
+      local entity, err = Entity.new(entity_schema)
+      if not entity then
+        return nil, fmt("schema of entity '%s' is invalid: %s", entity_name,
```

Comment:
> This would mean that there are errors that the MetaSchema won't catch, but Entity.new will?

Yes. The MetaSchema checks that it's a valid Schema, not that it's a valid Entity. This was already the case before this PR, we just didn't catch those errors. (I hit this during development when I wrote a bad entity).

> If so, maybe the MetaSchema should be responsible for validating such subclasses itself? This way we don't duplicate error handling here, which seems like a design flaw as it stands.

My initial thought about this is that a MetaSchema-based validation of Entity structures would be doable with a "MetaEntity" subschema for the MetaSchema, in which the additional constraints are specified (which would be conceptually a pretty cool solution). That would require a little structural tweaking, so I would prefer to implement this improvement on a further PR though.